### PR TITLE
Connected workspaces rewrites & updates for Beta release

### PR DESCRIPTION
**Pull request by cwarnermm**
Docs Impact:
- New [Connected Workspaces](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7575/onboard/connected-workspaces.html#connected-workspaces-beta) page replaces existing Shared Channels documentation. 
- Existing config settings [moved](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7575/configure/site-configuration-settings.html#connected-workspaces) from Experimental to Site Configuration.
- New config settings [added](http://mattermost-docs-preview-pulls.s3-website-us-east-1.amazonaws.com/7575/configure/site-configuration-settings.html#connected-workspaces) to Site Configuration

Documentation:
- https://github.com/mattermost/mattermost/pull/28920
- https://github.com/mattermost/mattermost/pull/28116
- https://github.com/mattermost/mattermost/pull/27820
- https://github.com/mattermost/mattermost/pull/28131
-  https://github.com/mattermost/mattermost/pull/28130
- https://github.com/mattermost/mattermost/pull/28020
- https://github.com/mattermost/mattermost/pull/28154

### CODE REVIEW

**Comment by cwarnermm** [comment]
@mgdelacroix - Please confirm whether the [remote clusters](https://docs.mattermost.com/configure/experimental-configuration-settings.html#exp-remoteclusters) configuration setting should remain documented as-is via Experimental, or if it needs changes for connected workspaces.

File: source/conf.py [hunk 1]
```diff
@@ -1508,7 +1508,7 @@ def setup(_: Sphinx):
 "configure/configuration-settings.html#user-status-away-timeout":
         "https://docs.mattermost.com/configure/experimental-configuration-settings.html#user-status-away-timeout",
 "configure/configuration-settings.html#enable-shared-channels":
-        "https://docs.mattermost.com/configure/experimental-configuration-settings.html#enable-shared-channels",
+        "https://docs.mattermost.com/configure/site-configuration-settings.html#enable-shared-channels",
```
Comment (wiggin77) [inline source/conf.py:1511]: Should this anchor still be called "enable-shared-channels" ?

Comment (cwarnermm) [reply source/conf.py:1511]: Nice catch, @wiggin77! Thank you!

Comment (cwarnermm) [inline source/conf.py:1511]:
```suggestion
        "https://docs.mattermost.com/configure/site-configuration-settings.html#enable-connected-workspaces",
```

Comment (wiggin77) [reply source/conf.py]: There are a few instances of this.

Comment (cwarnermm) [reply source/conf.py]: I believe all the remaining instances left are for the "from" portion of each page redirect, and not the defined destination. Are you seeing differently?

File: source/onboard/connected-workspaces.rst [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    ``/secure-connection status``
+
+    Status details include:
+
+    - Connection ID
+    - Connection URL
+    - Description
+    - Invite accepted (Yes/No)
+    - Online (Yes/No)
+    - Last ping timestamp (UTC)
```
Comment (mgdelacroix) [inline source/onboard/connected-workspaces.rst:224]: As part of the last batch of changes we added the `Deleted` column to the output of this command

```suggestion
    - Last ping timestamp (UTC)
    - Deleted
```

File: source/configure/site-configuration-settings.rst [hunk 3]
```diff
@@ -1715,3 +1716,71 @@ Enable end user notices
...
+  :configjson: ConnectedWorkspacesSettings.DisableSharedChannelsStatusSync
+  :environment: N/A
+
+  - **true**: **(Default)** Channel as well as channel member status and availability is synchronized at regular intervals.
+  - **false**: Channel as well as member status and availability isn't synchronized.
+
+Disable shared channel status sync
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Disable channel as well as member status and availability synchronization between connected workspaces.
```
Comment (mgdelacroix) [inline source/configure/site-configuration-settings.rst:1764]: This configuration parameter only affects member status sync, I think we can remove the channel reference here and in the rest of the instances

```suggestion
Disable member status and availability synchronization between connected workspaces.
```

**Review by mgdelacroix — CHANGES_REQUESTED** [review]
Looking really good! Added a couple of change proposals and questions

**Comment by mgdelacroix** [comment]
> Please confirm whether the [remote clusters](https://docs.mattermost.com/configure/experimental-configuration-settings.html#exp-remoteclusters) configuration setting should remain documented as-is via Experimental, or if it needs changes for connected workspaces.

We're maintaining those configuration parameters marked as deprecated, and they will have no effect beyond the initial automated migration to `ConnectedWorkspaces` the first time `v10.2` runs, so I think we should mark them as deprecated in the experimental section and point to the new ones.

> Also, please confirm whether any of the connected workspace configuration settings should be added to the [Telemetry](https://docs.mattermost.com/manage/telemetry.html#telemetry) page.

Yes, all `ConnectedWorkspaces` properties have been added to the telemetry payload with the following format:

```js
{
  "config_connected_workspaces": {
    "enable_shared_channels": val,
    "enable_remote_cluster_service": val,
    "disable_shared_channels_status_sync": val,
    "max_posts_per_sync": val,
  }
}
```

**Pull request merged**